transport/conn.go

```diff
@@ -424,9 +440,15 @@ func (c *Conn) Shard() int {
 	return int(c.shard)
 }
 
+func (c *Conn) deactivateConnClose() {
+	c.w.connClose = nil
+	c.r.connClose = nil
+}
+
```

Comment:
This is clearly a race.

Reply:
I suggest we use [sync.Once](https://pkg.go.dev/sync#example-Once) in Conn::Close to ensure the actual closing happens once.